arm/Microsoft.MachineLearningServices/workspaces/deploy.bicep

```diff
@@ -31,8 +34,8 @@ param associatedContainerRegistryResourceId string = ''
 @description('Optional. Specify the type of lock.')
 param lock string = 'NotSpecified'
 
-@description('Optional. Enables system assigned managed identity on the resource.')
-param systemAssignedIdentity bool = false
+@description('Optional. Identity for the resource.')
+param identity object = {}
```

Comment:
Is there a reason why you changed this? We use the same implementation (almost) across the board and if the current one does not serve all use cases, we should update the other ones too. cc: @MariusStorhaug

Reply:
@MrMCake  @MariusStorhaug The way it was implemented it supported only system assigned managed identity. This way it is possible to provide user assigned managed identities as well. I propose to keep it this way (or similar) and if necessary also change it in the other modules. What do you think?

Reply:
Could you check any other modules that supports user assigned identity, that would be where you see one way of supporting both. One thing we could work to improve, is the user assigned identity object that is created. Should really only be an array of strings.

Reply:
@MariusStorhaug I changed the implementation following your suggestion. Is it ok that way?

Reply:
Ye, the implementation seems similar to the other ones now. I see in the diff that when we did change this last time, the resource type did not support `userAssignedIdentities`, now it seems it does. Thank you for adding it and help keeping the code consistent across modules!